### PR TITLE
Fixes some issues encountered during model.generate invocations with do_sample=True

### PR DESCRIPTION
**Pull request by jploski**
See commit comments. I'm not 100% sure if the outputs.logits.squeeze(0) fix is correct or if it will create problems with generation using batch size > 1, but it certainly fixes a critical bug with batch size = 1.

### CODE REVIEW

File: retnet/modeling_retnet.py [hunk 1]
```diff
@@ -589,7 +589,7 @@ def generate(
         bos_token_id=0,
         eos_token_id=0,
         do_sample=False,
-        top_k=1,
+        top_k=0,
```
Comment (syncdoth) [inline retnet/modeling_retnet.py:592]: I checked huggingface/transformers repo, and found their implementation has `top_k=50` as default. Still, I agree that setting `top_k=0` in this implementation makes more sense. Thanks!

File: retnet/modeling_retnet.py [hunk 2]
```diff
@@ -636,7 +636,8 @@ def generate(
                            use_cache=True,
                            return_dict=True,
                            sequence_offset=prompt_len + i)
-            logit = outputs.logits
+            logit = outputs.logits.squeeze(0)
```
Comment (syncdoth) [inline retnet/modeling_retnet.py:639]: This is not the correct way to go. The logits shape should be `[batch_size, seq_len, vocab_size]`, so if the batch size is not 1, this line does nothing.

Instead, we should take the last token's logit only, by setting

```
logit = outputs.logits[:, -1]  # [batch_size, vocab_size]
```

This will solve the problem.

File: retnet/modeling_retnet.py [hunk 3]
```diff
@@ -636,7 +636,8 @@ def generate(
...
+            past_key_values = outputs.past_key_values
```
Comment (syncdoth) [inline retnet/modeling_retnet.py:640]: That's right, I forgot this! Thx

**Review by syncdoth — CHANGES_REQUESTED** [review]
Thanks for the contribution! I have left some comments, could you please consider them?

File: retnet/modeling_retnet.py (hunk 2 above)
Comment (jploski) [reply retnet/modeling_retnet.py]: Right, I committed a change, using outputs.logits[:, -1, :] to make it a bit more explicit.

**Review by syncdoth — APPROVED** [review]
LGTM. Thanks for your contribution!
